### PR TITLE
Firewall and hub and Spoke implementation

### LINKED ISSUE #184: As a network administrator I want all the network traffic to be regulated to ensure security
As a network administrator I want all the network traffic to be regulated to ensure security 

**Note**: Use sample 2

# Acceptance Criteria 
* A Hub(DMZ) Vnet is created with a Azure firewall service configured to protect it
* All ports and IPs required for ADB control plane are opened
* All other ports and IPs are blocked by default

# Definition of done
* ADB control plane is able to communicate with the workspace
* External connections to the ADB Vnet is blocked/regulated

Depends on #146

### PR DESCRIPTION
**Pull request by namitms**
## Purpose
* Firewall ARM template
* Routing tables
* Vnet peering

## Does this introduce a breaking change?
- [x] Yes
- [ ] No

## Author pre-publish checklist:
- [ ] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [ ] Made corresponding changes to the documentation

## Closes #184

### CODE REVIEW

**Comment by xinsnake** [comment]
Hi @namitms, I personally prefer not to have the reference of "lumos" in the variable name. Otherwise, happy for you to merge.

**Review by xinsnake — CHANGES_REQUESTED** [review]
Please remove the reference to "lumos" :)

**Comment by namitms** [comment]
> Please remove the reference to "lumos" :)

Lumos removed :)

**Review by xinsnake — APPROVED** [review]

File: single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh [hunk 1]
```diff
@@ -172,27 +214,59 @@ if [[ -z $nsg_arm_output ]]; then
     exit 1
 fi
 
+echo "Deploying Route Table..."
+nsg_arm_output=$(az deployment group create \
```
Comment (tomomi-code) [inline single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh:218]: This is not `nsg` so better to change the variable name I guess.

File: single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh [hunk 2]
```diff
@@ -172,27 +214,59 @@ if [[ -z $nsg_arm_output ]]; then
...
+        routeTableName="$routeTableName" \
     --output json)
 
 if [[ -z $vnet_arm_output ]]; then
     echo >&2 "Virtual Network ARM deployment failed."
     exit 1
 fi
 
+echo "Deploying Firewall..."
+nsg_arm_output=$(az deployment group create \
```
Comment (tomomi-code) [inline single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh:249]: Here as well.

**Comment by tomomi-code** [comment]
In `deploy.sh`, we have `Summary or resources created:` section at the bottom. So please add the resources' detail which you added in this PR.

File: single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh (hunk 1 above)
Comment (namitms) [reply single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh]: Fixed

File: single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh (hunk 2 above)
Comment (namitms) [reply single_tech_samples/databricks/sample2_enterprise_azure_databricks_environment/deploy.sh]: Fixed

**Comment by namitms** [comment]
> In `deploy.sh`, we have `Summary or resources created:` section at the bottom. So please add the resources' detail which you added in this PR.

This is fixed

**Review by tomomi-code — APPROVED** [review]
Thanks for your updating! LGTM :)

**Pull request merged**